lms/djangoapps/edraak_marketing_email/tests/test_views.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework.test import APITestCase
+from rest_framework import status
+from django.urls import reverse
+from student.tests.factories import UserFactory
+from edraak_marketing_email.helpers import (
+    subscribe_to_marketing_emails,
+    unsubscribe_from_marketing_emails
+)
+
+
+class TestUnsubscribeUserAPIView(APITestCase):
+    """Tests for handling edraak marketing email unsubscription"""
+
+    def setUp(self):
+        super(TestUnsubscribeUserAPIView, self).setUp()
+        self.client = self.client_class()
+        self.url = reverse('edraak_marketing_email:unsubscribe')
+
+    def test_user_unsubscribed_successfully(self, *args):  # pylint: disable=unused-argument
+        user = UserFactory()
```

Comment:
```suggestion
        user = UserFactory()
        UnsubscribedUser.objects.create(user=user)
```

Reply:
There is another test to test the already unsubscribed users so no need to add it here.

Reply:
no problem